src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1135,6 +1136,53 @@ private void RunConnectRequestHandlerTask(ConnectParams connectParams)
             }).ContinueWithOnFaulted(null);
         }
 
+        /// <summary>
+        /// Handle new change password requests
+        /// </summary>
+        /// <param name="connectParams"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        protected async Task HandleChangePasswordRequest(
+            ChangePasswordParams changePasswordParams,
+            RequestContext<PasswordChangeResponse> requestContext)
+        {
+            Logger.Write(TraceEventType.Verbose, "HandleChangePasswordRequest");
+            PasswordChangeResponse newResponse = new PasswordChangeResponse();
+            try
+            {
+                RunChangePasswordRequestHandlerTask(changePasswordParams);

```

Comment:
does this run on the dispatcher thread?  if so, it should be wrapped in a task or otherwise made truly async.

Reply:
It does not run on the dispatcher thread, therefore I'm going to rename it.